modules/azure/event_grid_topic_subscription/variables.tf

```
@@ -48,3 +48,20 @@ variable "delivery_properties" {
   }))
   description = "parameters for delivery properties, set property_value if type is static, otherwise use source_field for dynamics"
 }
+
+variable "advanced_filter" {
+  type = list(object({
+    string_begins_with = object({
+      values = list(string)
+      key    = string
+    })
+    string_ends_with = object({
+      values = list(string)
+      key    = string
+    })
```

```suggestion
    string_begins_with = optional(object({
      values = list(string)
      key    = string
    }), [])
    string_ends_with = optional(object({
      values = list(string)
      key    = string
    }), [])
```

Please make them optional

---

Sure thanks